share/spack/setup-env.sh

```diff
@@ -236,9 +236,16 @@ fi;
 #
 # Set up modules and dotkit search paths in the user environment
 #
-_sp_sys_type=$(spack-python -c 'print(spack.architecture.sys_type())')
-_sp_dotkit_root=$(spack-python -c "print(spack.util.path.canonicalize_path(spack.config.get_config('config').get('module_roots', {}).get('dotkit')))")
-_sp_tcl_root=$(spack-python -c "print(spack.util.path.canonicalize_path(spack.config.get_config('config').get('module_roots', {}).get('tcl')))")
+
+_python_command=$(printf  "%s\\\n%s\\\n%s" \
```

Comment:
`$()` is a bashism, and `zsh` understands it too.  It's easy to replace with ``.  Let's do that later though -- the script doesn't work with `dash` or plain `sh` yet anyway.  I think this PR is great!

Reply:
According to http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_03, `$()` is posix.

Reply:
And wouldn't `_python_command=$(printf  '%s\n%s\n%s'` work as well?

Reply:
>  Let's do that later though -- the script doesn't work with dash or plain `sh` yet anyway. I think this PR is great!

@tgamblin  The PR is good to go as far as I am concerned

Reply:
> And wouldn't _python_command=$(printf '%s\n%s\n%s' work as well?

@junghans Nope. If I don't double quote `zsh` expands the new lines, which causes `spack-python -c ...` to fail later.